src/pytorch_lightning/overrides/base.py

```diff
@@ -54,29 +55,47 @@ def forward(self, *args: Any, **kwargs: Any) -> Any:
 
 
 class _LightningModuleWrapperBase(DeviceDtypeModuleMixin, torch.nn.Module):
-    def __init__(self, pl_module: Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase]) -> None:
+    def __init__(
+        self, forward_module: Optional[Union["pl.LightningModule", _LightningPrecisionModuleWrapperBase]]
+    ) -> None:
         """Wraps the user's LightningModule and redirects the forward call to the appropriate method, either
         ``training_step``, ``validation_step``, ``test_step``, or ``predict_step``.
 
         Inheriting classes may also modify the inputs or outputs of forward.
 
         Args:
-            pl_module: the model to wrap
+            forward_module: The module to wrap. If it's not a LightningModule, it must have an attribute ``.module``
+                pointing to a LightningModule reference.
         """
         super().__init__()
-        self.module = pl_module
+        if not isinstance(forward_module, pl.LightningModule) and (
+            not isinstance(getattr(forward_module, "module", None), pl.LightningModule)
+        ):
+            raise ValueError(
+                "`forward_module` must be a `LightningModule` instance or have an attribute `.module` pointing to one,"
+                f" got: {forward_module.__class__.__qualname__}"
+            )
+        # TODO: In v1.10, remove the Optional type from forward_module and remove the assertion
```

Comment:
can remove the optional now by making an assertion in the child classes.

Reply:
This is the least ugliest way I found. I can't express what you want in mypy. It is not smart enough to resolve an assertion like this:

```
        assert not (pl_module is None and forward_module is None)
        reveal_type(forward_module)
```

`src/pytorch_lightning/overrides/data_parallel.py:72: note: Revealed type is "Union[pytorch_lightning.core.module.LightningModule, pytorch_lightning.overrides.base._LightningPrecisionModuleWrapperBase, None]"
`

I don't think this is worth investing time. It will be very easy to resolve this in 1.10 by just removing the code.

Reply:
cool!